demos/test-app/lib.rs

```diff
@@ -106,160 +74,108 @@ pub enum AlternativeOriginsMode {
 pub fn http_request(req: HttpRequest) -> HttpResponse {
     let parts: Vec<&str> = req.url.split('?').collect();
     let path = parts[0];
-    let mut headers = vec![];
-    headers.push(("Access-Control-Allow-Origin".to_string(), "*".to_string()));
-    let certificate_header =
-        ASSET_HASHES.with(|a| make_asset_certificate_header(&a.borrow(), path));
 
     match path {
-        "/.well-known/ii-alternative-origins" => ALTERNATIVE_ORIGINS_MODE.with(|m| {
-            let mode = m.borrow();
-            let mut status_code = 200;
-            match mode.clone() {
+        ALTERNATIVE_ORIGINS_PATH => ALTERNATIVE_ORIGINS_MODE.with_borrow(|mode| {
+            let mut certified_response = certified_ok_response(path, req.certificate_version)
+                .expect("/.well-known/ii-alternative-origins must be certified");
+            match mode {
                 CertifiedContent => {
-                    headers.push(certificate_header);
+                    // don't tamper with the certified response
                 }
-                Redirect { location } => {
-                    // needs to be certified content for the service worker
-                    // (which the browser will then ignore and redirect anyway)
-                    headers.push(certificate_header);
-                    headers.push(("Location".to_string(), location));
-                    status_code = 302;
+                UncertifiedContent => {
+                    // drop the IC-Certificate and IC-Certificate-Expr headers
+                    certified_response
+                        .headers
+                        .retain(|(header_name, _)| !header_name.to_lowercase().starts_with("ic-"));
```

Comment:
nit: maybe this filter is a bit too generous?  How about changing to `.starts_with("ic-certificate")`?